### PR TITLE
fix armor pierce on uniques and display affix values

### PR DESCRIPTION
**Pull request by qndel**
Resolves https://github.com/diasurgical/devilutionX/issues/4044
And displays the actual armor pierce value when not in hellfire

### CODE REVIEW

**Comment by AJenbo** [comment]
Isn't this a duplicate of what is happening here? 

https://github.com/diasurgical/devilutionX/blob/master/Source/items.cpp#L1090

**Comment by qndel** [comment]
> Isn't this a duplicate of what is happening here?
> 
> https://github.com/diasurgical/devilutionX/blob/32d39c65246b17fbf6b89d4a914beb04f02ee7e8/Source/items.cpp#L1090

yeah, the old one should be removed I think, will test later :D

**Comment by AJenbo** [comment]
needs testing = draft

File: Source/items.cpp [hunk 1]
```diff
@@ -710,8 +710,15 @@ int CalculateToHitBonus(int level)
 	}
 }
 
-int SaveItemPower(Item &item, const ItemPower &power)
+int SaveItemPower(Item &item, ItemPower &power)
 {
+	if (!gbIsHellfire) {
+		if (power.type == IPL_TARGAC) {
+			power.param1 = 1 << power.param1;
```
Comment (DakkJaniels) [inline Source/items.cpp:717]: I know these bitshifts were here before, but looking at piercing, puncturing, and bashing, this results in 2-8, 4-16, and 6-24, which doesn't match the original values.
image

Comment (qndel) [reply Source/items.cpp:717]: quite the opposite
3 << 1 = 6
3 << 2 = 12
3 << 3 = 24

Comment (DakkJaniels) [reply Source/items.cpp:717]: oops sorry about that, saw the parameters backwards ( `param1 << 1' etc ) by accident.

**Review by StephenCWills — APPROVED** [review]
I know this is still in draft, but I tried generating a few rings and it seems to work properly.

**Pull request merged**